marklogic-data-hub-central/ui/src/components/modeling/graph-view/graph-vis/graph-vis.tsx

```diff
@@ -102,35 +152,75 @@ const GraphVis: React.FC<Props> = (props) => {
     if (graphType === "shape") {
       nodes = props.entityTypes && props.entityTypes?.map((e) => {
         let label = "";
-        return {
-          ...graphConfig.defaultNodeProps,
-          id: e.entityName,
-          label: label.concat(
-            "<b>", e.entityName, "</b>\n",
-            "<code>", getNumInstances(e.entityName).toString(), "</code>"
-          ),
-          title: e.entityName + " tooltip text", // TODO use entity description
-          color: {
-            background: getColor(e.entityName),
-            border: getColor(e.entityName),
-          },
-          chosen: {
-            node: function(values, id, selected, hovering) {
-              if (selected && hovering) {
-                values.color = graphConfig.nodeStyles.hoverColor;
-                values.borderColor = graphConfig.nodeStyles.selectColor;
-                values.borderWidth = 3;
-              } else if (selected) {
-                values.color = getColor(id);
-                values.borderColor = graphConfig.nodeStyles.selectColor;
-                values.borderWidth = 3;
-              } else if (hovering) {
-                values.color = graphConfig.nodeStyles.hoverColor;
-                values.borderWidth = 0;
+        if (e.entityName === modelingOptions.selectedEntity && props.entitySelected) {
```

Comment:
There is a lot of duplicated code here. I would refactor like this:

const node = {ALL THE PROPERTIES CURRENTLY DUPLICATED}
if (e.entityName === modelingOptions.selectedEntity && props.entitySelected) {
  // ADD SPECIAL BORDER/PHYSICS TO node OBJECT (and anything else that is special)
}

Reply:
Refactored and removed the duplicate code.